# HG changeset patch
# Parent e7a2396134ad9dbe6355375f9864bfd9aa3725b7
Add cmake target "apidoc" for creating doxygen API documentation

Also include a minimal Doxyfile for creating include/apidoc/

diff --git a/include/CMakeLists.txt b/include/CMakeLists.txt
--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -3,3 +3,7 @@
 INSTALL(FILES ${headers}
         DESTINATION include/polarssl
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+
+ADD_CUSTOM_TARGET(apidoc
+  COMMAND doxygen Doxyfile
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/include/Doxyfile b/include/Doxyfile
new file mode 100644
--- /dev/null
+++ b/include/Doxyfile
@@ -0,0 +1,4 @@
+PROJECT_NAME           = PolarSSL
+INPUT                  = polarssl
+HTML_OUTPUT            = apidoc
+GENERATE_LATEX         = NO
